packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -317,7 +318,32 @@ export class SummaryManager extends TypedEventEmitter<ISummaryManagerEvents> imp
         }
 
         if (delayMs > 0) {
-            await delay(delayMs);
+            // eslint-disable-next-line prefer-const
+            let timer;
+            let resolveFn;
+            // Create a listener that will break the delay if we've exceeded the initial delay ops count.
+            const opsListenerFn = () => {
+                if (this.summaryCollection.opsSinceLastAck >= this.opsToBypassInitialDelay) {
+                    clearTimeout(timer);
+                    resolveFn();
+                }
+            };
+            // Create a Promise and capture its resolve function.
+            const delayPromise = new Promise<void>((resolve) => { resolveFn = resolve; }).catch(() => {
+                this.summaryCollection.removeOpListener(opsListenerFn);
```

Comment:
no need to clear opsListenerFn here as it's always done on line 346 (delayPromise itself never throws)

Reply:
Also this code in catch block will never run, as you do not capture second callback ("reject") and resolveFn() is never called with a promise (that might be rejected), thus there is absolutely no way for this promise to be rejected.

Reply:
Cool. Thanks.